### ISSUE TITLE
Astro-critters 2.1 completely broke it?

### ISSUE THREAD

**Issue by Steffan153** [PlayForm/Inline#235]
It just errors (same on Netlify build):

![image](https://github.com/astro-community/AstroCritters/assets/40404519/b30ff699-d70f-4aec-a7c4-9d60c1627a4d)

Downgrading to v2.0.11 fixes it instantly. Astro 4.4.4.

**Comment by NikolaRHristov** [comment]
Some files cannot be inlined. You can try to debug by plugging into the `Wrote` hook on line: https://github.com/astro-community/AstroCritters/blob/7ed20967726f25dca599eea8e32f0b1ac4a425a0/Source/Function/Integration.ts#L82

Something like:

```ts
await (
	await (
		await (
			await new (await import("files-pipe")).default(
				Cache,
				Logger
			).In(Path)
		).By("**/*.html")
	).Not(Exclude)
).Pipe(
	Merge(Action, {
		Wrote: async ({ Buffer }) => {
			try {
				_Critters.process(Buffer.toString());
			} catch (_Error) {
				console.log(_Error);
			}
		},
	} satisfies Action)
);
```
,

should work.

**Comment by Steffan153** [comment]
I couldn't get any errors to show up with that method, but I added an `Action: { Failed: ... }` to my integration settings and here's the error I got:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "paths[0]" argument must be of type string. Received an instance of URL
    at validateString (node:internal/validators:162:11)
    at Object.resolve (node:path:1101:7)
    at Critters.getCssAsset (file:///[censored]/node_modules/critters/dist/critters.mjs:1205:27)
    at Critters.embedLinkedStylesheet (file:///[censored]/node_modules/critters/dist/critters.mjs:1268:30)
    at file:///[censored]/node_modules/critters/dist/critters.mjs:1132:57
    at Array.map (<anonymous>)
    at Critters.process (file:///[censored]/node_modules/critters/dist/critters.mjs:1132:40)
    at Wrote (file:///[censored]/node_modules/astro-critters/Target/Function/Integration.js:1:773)
    at Module.l (file:///[censored]/node_modules/files-pipe/Target/Function/Pipe.js:1:257)
    at async o.Pipe (file:///[censored]/node_modules/files-pipe/Target/Class/Files.js:1:421) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

**Comment by Steffan153** [comment]
If I call `.toString` on the `path` option to Critters, then the errors go away, but then it can't find the stylesheets because they get joined wrong:
```
Time 0.792666
Unable to locate stylesheet: /Users/[...]/[...]/file:/Users/[...]/[...]/dist/_astro/[...]
```

**Comment by Steffan153** [comment]
Converting the URL to a path with `fileURLToPath` like it was in v2.0 fixes it:
https://github.com/astro-community/AstroCritters/blob/6dfe278f0e1fa7095273a253062e5848cbfda8e5/Source/Index.ts#L51-L59

**Comment by NikolaRHristov** [comment]
Implementing, will be fixed in `v2.1.2`.

**Comment by NikolaRHristov** [comment]
Fixed in https://github.com/astro-community/AstroCritters/releases/tag/v2.1.2, thank you so much for reporting and debugging this!
